### PR TITLE
Fixing issue#25 and invalid_args_issues

### PR DESCRIPTION
**Pull request by gbenroscience**
Reset the master to workaround 26 and reapplied the fix to a branch which I have used to create this PR.
Please check

### CODE REVIEW

File: src/main/java/parser/DataSetFormatter.java [hunk 1]
```diff
@@ -202,8 +202,24 @@ public final void scanCommaSeparatedData(String myStr) {
     }
 
     public static void main(String args[]) {
-        String func = "sum(@(x)x^3+2*x+1,3,@(x)cos(x+4),5,-6,7,-8,8)";//"sort(sin(2),cosh(4),-6,log(12,10),-12,2*sin(3),cos(4),34)";
+        //String func = "sum(@(x)x^3+2*x+1,3,@(x)cos(x+4),5,-6,7,-8,8)";//"sort(sin(2),cosh(4),-6,log(12,10),-12,2*sin(3),cos(4),34)";
```
Comment (judovana) [inline src/main/java/parser/DataSetFormatter.java:199]: All the new  sum/sin expressions belongs to the https://github.com/gbenroscience/ParserNG/blob/master/src/test/java/parser/methods/ext/SumTest.java
If you build this PR on https://github.com/gbenroscience/ParserNG/pull/23 you will get even more tests

**Comment by judovana** [comment]
As I can read the fix, it is still workaround, not fix. In addition, it is killing many tests. They fall carrectly. You have caused fixed the issue https://github.com/gbenroscience/ParserNG/issues/25 but introduced more severe bug -  any statistical function with more then one argument, now got them cumulated, instead of passed as individual tokens. eg for 
`sum(1,2,3)`  or `prod(1,2,3)` engine no longer passes in three tokens of 1,2,3 but one token 123. That would be much more severe regression then is  issue#25.

I would reocmend to change commit/pr title to "fixing issue#25 - allowing statistical functions to accept single expression as parameter".
Also I would recomend to introduce body of commit message, saying what the change does.

Still this is irrelevant, due to the much more severe issue thei PR is casuing.

File: src/main/java/parser/DataSetFormatter.java (hunk 1 above)
Comment (judovana) [reply src/main/java/parser/DataSetFormatter.java:199]: Also all the old code from main can be moved to to tests. But I can do this  later for you if you wish.

**Comment by gbenroscience** [comment]
If you can check this asap, I would be glad as I have taken off so much time to work on this fix.

**Comment by judovana** [comment]
Well all checks have passed. The fix is mixing reformatting, refactoring and logical changes, but I can see why you you did it.
I had rebased the sinSum bracnh whcih have the main tests, and it seesm you have fixed both `prod(1+1)` and `sin(1+1,1+1)` issue

**Comment by gbenroscience** [comment]
So we are cool, right?

**Comment by judovana** [comment]
I'm working on it. Pls few more time.

**Comment by judovana** [comment]
For now I had rebased sin and prod PR to be based on you, and extracted the main method of yours to tests. It is green. And that is something!

Now I'm rebasing the rounding PR, as that was the original place which hit the issue.

**Comment by judovana** [comment]
I had rebased the rounding pr and completely removed the workround. All is passing.
Please merge this one. Good job!

if you will be so kind and merge also  rounding pr will be awesome.  You can merge also sin and prod tests, although they seem to have some concurency issue on windows (I will elaborate if it will persists).
Other PRs are ok to go too, but are not related to this issue. (will rebase them once you merge)

**Pull request merged**